src/inventorymanager.cpp

```diff
@@ -842,7 +842,8 @@ void ICraftAction::apply(InventoryManager *mgr,
 			count_remaining--;
 
 		// Get next crafting result
-		found = getCraftingResult(inv_craft, crafted, temp, false, gamedef);
+		if (!getCraftingResult(inv_craft, crafted, temp, false, gamedef))
+			break;
 		PLAYER_TO_SA(player)->item_CraftPredict(crafted, player, list_craft, craft_inv);
 		found = !crafted.empty();
```

Comment:
Probably would be a good idea to simplify this loop's condition by removing the superfluous "found" variable.